fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/temp/AliasMap.java

```diff
@@ -211,6 +211,11 @@ public boolean containsTarget(@Nonnull final CorrelationIdentifier alias) {
         return map.containsValue(alias);
     }
 
+    public boolean containsMapping(@Nonnull final CorrelationIdentifier source,
+                                   @Nonnull final CorrelationIdentifier target) {
+        return containsSource(source) && containsTarget(target) && getTargetOrThrow(source).equals(target);
```

Comment:
The `OrThrow` cannot really throw because we already checked that the source is present, but looks a little odd in a predicate. Wouldn't `target.equals(getTarget(source))` do everything we need?

Reply:
It does except the return of that is `Nullable` which causes a warning. I could wrap that with a `Objects.requreNotNull()` but this is doing the same. So it's an artifact of shutting up the compiler.